### PR TITLE
PWN-6258 - Add Sell Lock and Sell Payload Fragment, also create custo…

### PR DESCRIPTION
**Pull request by davran1001**
…m input view

## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6258

### CODE REVIEW

**Comment by 1sakov** [comment]
screenshots please

File: app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestoreFragment.kt [hunk 1]
```diff
@@ -120,6 +121,7 @@ class CommonRestoreFragment :
         requireActivity().onBackPressedDispatcher.addCallback(viewLifecycleOwner) {
             popAndReplaceFragment(OnboardingRootFragment.create(), inclusive = true)
         }
+        replaceFragment(SellPayloadFragment.create())
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestoreFragment.kt:124]: what is this?

File: app/src/main/java/org/p2p/wallet/sell/SellModule.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import org.p2p.wallet.sell.ui.lock.SellLockContract
+import org.p2p.wallet.sell.ui.lock.SellLockPresenter
+import org.p2p.wallet.sell.ui.payload.SellPayloadContract
+import org.p2p.wallet.sell.ui.payload.SellPayloadPresenter
+
+object SellModule : InjectionModule {
+
+    override fun create(): Module = module {
+        single { SellPayloadPresenter() } bind SellPayloadContract.Presenter::class
+        single { SellLockPresenter() } bind SellLockContract.Presenter::class
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/sell/SellModule.kt:16]: why presenters are single?

File: app/src/main/java/org/p2p/wallet/sell/ui/lock/SellLockFragment.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        fun create() = SellLockFragment()
+    }
+
+    override val presenter: SellLockContract.Presenter by inject()
+    private val binding: FragmentSellLockBinding by viewBinding()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        with(binding) {
+        }
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/sell/ui/lock/SellLockFragment.kt:25]: why it's empty?
If your PR is not ready, please mark it with a label

File: app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadFragment.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    override val presenter: SellPayloadContract.Presenter by inject()
+    private val binding: FragmentSellPayloadBinding by viewBinding()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        with(binding) {
+        }
+    }
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadFragment.kt:26]: same here

**Review by 1sakov — CHANGES_REQUESTED** [review]
looks like this PR is not ready yet

File: app/src/main/java/org/p2p/wallet/sell/ui/lock/SellLockFragment.kt (hunk 3 above)
Comment (davran1001) [reply app/src/main/java/org/p2p/wallet/sell/ui/lock/SellLockFragment.kt:25]: Yeah design was changed 3 times, while i was developing this, we will discuss about this today on standup
That's why PR was not finished.

File: app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+    private suspend fun isUserHasTransactionsInProcess(): Boolean {
+        val userTransactions = sellInteractor.loadUserSellTransactions()
+        return userTransactions.isNotEmpty() && userTransactions.all {
+            it.status == MoonpaySellTransaction.TransactionStatus.COMPLETED
+        }
+    }
+
+    private suspend fun initView() {
+        val solToken = solTokenProvider.getUserSolanaToken() ?: return
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt:44]: don't forget to add logic if user doesn't have SOL

File: app/src/main/res/layout/fragment_sell_payload.xml [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    <org.p2p.uikit.components.UiKitEditText
+        android:id="@+id/editTextToken"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginHorizontal="16dp"
+        android:layout_marginTop="120dp"
+        app:layout_constraintTop_toBottomOf="@id/toolbar" />
+
+    <org.p2p.uikit.components.UiKitEditText
+        android:id="@+id/editTextAmount"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_sell_payload.xml:50]: editTextInputUsd

File: app/src/main/res/layout/fragment_sell_payload.xml [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+        android:textAppearance="@style/UiKit.TextAppearance.Regular.Label1"
+        android:textColor="@color/text_sky"
+        app:layout_constraintBottom_toBottomOf="@id/textViewSell"
+        app:layout_constraintStart_toEndOf="@id/textViewSell"
+        app:layout_constraintTop_toTopOf="@id/textViewSell"
+        tools:text="all 4.53 SOL" />
+
+
+    <org.p2p.uikit.components.UiKitEditText
+        android:id="@+id/editTextToken"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_sell_payload.xml:42]: editTextInputToken

File: app/src/main/res/layout/fragment_sell_payload.xml [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        android:layout_marginTop="1dp"
+        android:background="@drawable/bg_app_bottom_rounded"
+        android:padding="16dp"
+        android:textAppearance="@style/UiKit.TextAppearance.Regular.Label1"
+        android:textColor="@color/text_mountain"
+        app:layout_constraintTop_toBottomOf="@id/textViewRate"
+        tools:text="Included fee 0.6 SOL" />
+
+    <org.p2p.uikit.components.UiKitButton
+        android:id="@+id/buttonSend"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_sell_payload.xml:85]: buttonCashOut?

File: app/src/main/res/layout/fragment_sell_payload.xml [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+        style="@style/UiKit.Components.Button.Large"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_margin="16dp"
+        android:text="@string/common_cash_out"
+        app:layout_constraintBottom_toBottomOf="parent" />
+
+    <FrameLayout
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_sell_payload.xml:93]: why do you need FrameLayout?

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -613,6 +616,17 @@
...
     <string name="recovery_kit_logout_title"> Do you want to see your seed phrase?</string>
     <string name="recovery_kit_logout_message">You have to log out and log in again.</string>
+    <string name="sell_lock_title">Please, send crypto to\nMoonPay address</string>
+    <string name="sell_lock_message">To finish processing your request you need to send SOL to the address in the description. After 7 days this transaction will be automatically declined.</string>
+    <string name="sell_lock_button_action">Send SOL in Key App</string>
+    <string name="sell_lock_button_cancel">Remove from history</string>
+    <string name="sell_payload_title">Cash out token</string>
+    <string name="sell_all_sol">all %f SOL</string>
+    <string name="common_sell_all">Sell all</string>
+    <string name="common_sorry">Sorry</string>
```
Comment (gslevinkov) [inline app/src/main/res/values/strings.xml:626]: put common to common_strings in the top

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitEditText.kt [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    fun setHint(hint: String) {
+        binding.textViewHint.text = hint
+    }
+
+    val text: Editable?
+        get() = binding.editText.text
+
+    val stringText: String?
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitEditText.kt:81]: you can return emptyString instead of null

**Comment by gslevinkov** [comment]
it's better to merge this PR ASAP and implement logic and comment fixes in other branch

File: app/src/main/java/org/p2p/wallet/sell/ui/error/SellErrorFragment.kt [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+import org.p2p.wallet.swap.ui.orca.OrcaSwapFragment
+import org.p2p.wallet.utils.args
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.replaceFragment
+import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.utils.withArgs
+
+private const val ARG_ERROR_STATE = "ARG_ERROR_STATE"
+
+class SellErrorFragment : BaseFragment(R.layout.fragment_sell_error) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/sell/ui/error/SellErrorFragment.kt:17]: and again, this screen is out of scope of this task

you have https://p2pvalidator.atlassian.net/browse/PWN-6277 for corner cases

File: app/src/main/res/layout/fragment_sell_payload.xml (hunk 8 above)
Comment (davran1001) [reply app/src/main/res/layout/fragment_sell_payload.xml:85]: it was send, when I created it. After they renamed the button

File: app/src/main/res/layout/fragment_sell_payload.xml (hunk 9 above)
Comment (davran1001) [reply app/src/main/res/layout/fragment_sell_payload.xml:93]: to cover screen, cause otherwise I need to create secondary viewGroup for content. and hide it when shimmer is visible

File: app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/BuyTransactionDetailsBottomSheet.kt [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+
+import android.os.Bundle
+import android.view.View
+import org.koin.android.ext.android.inject
+import org.p2p.wallet.common.mvp.BaseMvpBottomSheet
+import org.p2p.wallet.databinding.DialogSendTransactionDetailsBinding
+import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.R
+
+class BuyTransactionDetailsBottomSheet : BaseMvpBottomSheet<
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/BuyTransactionDetailsBottomSheet.kt:11]: BuyTransactionDetailsBottomSheet

Buy??

File: app/src/main/res/layout/dialog_send_transaction_details.xml [hunk 14]
```diff
@@ -0,0 +1,178 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (gslevinkov) [inline app/src/main/res/layout/dialog_send_transaction_details.xml:1]: dialog_sell_transaction_details

**Review by gslevinkov — APPROVED** [review]
